flux-tests/tests/neg/error_messages/dfn_err.rs

```diff
@@ -0,0 +1,20 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+#![feature(custom_inner_attributes)]
+#![flux::dfn(nat(x: int) -> bool { 0 <= x })]
```

Comment:
I prefer `def` over `dfn`. Mentally, I parse `dfn` as an acronym. But no strong feelings about this one.

note: For a while, I've been feeling that it wasn't such a good idea to have separate attributes for everything. Maybe `flux::sig`,  `flux::field`, `flux::variant` and `flux::refined_by` work, but for these top-level definitions we could have a single attribute and parse different items, e.g.,

```rust
#[flux {
    fn nat(x: int) -> bool {
        x >= 0
    }
    
    fn uninterpreted(x: int) -> bool;
    
    qualif MyQ1(x: int, a: int)  {
        x == a + FORTY_TOO
    }
}]
```

Reply:
interesting thought! It does seem a _lot_ cleaner. 

btw, if you saw the `date.rs` test there is a bunch of "duplication" so I wonder if instead we should use a macro like Prusti so:

```rust
flux::def! { fn nat(x: i32) -> bool { 0 <= x } } 
```

Generates both the flux-level `def` and the rust-level `fn` ?

Something to ponder...

Reply:
Ah, yes I think I picked `dfn` vs `ufn` but I just realized your point is you can just use `fn` and disambiguate based on presence of definition! Definitely better.